src/Tools/dotnet-format/MSBuild/MSBuildCoreLoader.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// LICENSING NOTE: The license for this file is from the originating 
+// source and not the general https://github.com/dotnet/roslyn license.
+// See https://github.com/Microsoft/MSBuildLocator/blob/6631a6dbf9be72b2426e260c99dc0f345e79b8e5/src/MSBuildLocator/MSBuildLocator.cs
```

Comment:
IIRC, that PR hit some significant roadblocks. Have you reached out to the right folks on the .NET Core SDK and MSBuild teams to determine whether the solution here is the right one? I think the MSBuild Locator PR has pretty much stalled out. At a minimum, I'd start with @nguerrera.

Reply:
There were really two things with the locator PR:

1. It doesn't work on Mac (nor all flavors of Linux). This is a technical issue about how it is assuming it can P/Invoke to already loaded hostfxr.dll. This does not apply to the alternate approach of parsing dotnet -info.

2. It ties the locator to implementation details of the CLI. This applies to both approaches. See https://github.com/Microsoft/MSBuildLocator/pull/33#issuecomment-407240031

As on the locator PR, I can let (2) go here if you agree to update this code if/when the layout changes. We do not consider the current placement of msbuild dlls in the CLI as a contract for tools outside the CLI.